src/parser/mod.rs

```diff
@@ -9604,7 +9604,13 @@ impl<'a> Parser<'a> {
             top = Some(self.parse_top()?);
         }
 
-        let projection = self.parse_projection()?;
+        let projection = if dialect_of!(self is PostgreSqlDialect | GenericDialect)
```

Comment:
could we use a dialect method for the feature flag? so that this becomes e.g. `if self.dialect.supports_empty_projections() && self.peek_keyword(Keyword::FROM)`